fabric-content-registries-v0/src/main/java/net/fabricmc/fabric/api/util/OxidizableFamily.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.util;
+
+import static net.minecraft.block.Oxidizable.OxidizationLevel;
+
+import java.util.HashMap;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.ImmutableBiMap;
+import com.google.common.collect.ImmutableMap;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.Oxidizable;
+
+/**
+ * Represents a 'family' of blocks that can be affected by Oxidization.
+ *
+ * <p>Like vanilla's Copper Blocks, these come in four levels of Oxidization;
+ * Unaffected, Exposed, Weathered, and Oxidized.
+ *
+ * <p>Each block also has a 'Waxed' form.
+ */
+public record OxidizableFamily(
```

Comment:
Do you want this to have a public constructor? If not, I wouldn't use a record.

Reply:
Having a public constructor is technically fine, though it does circumvent the null check that the Builder does. I suppose I can change it to a regular class.

Reply:
you can write a canonical ctor that validates arguments, too